pennylane/transforms/metric_tensor.py

```diff
@@ -18,13 +18,25 @@
 import functools
 import warnings
 
-import numpy as np
 import pennylane as qml
+from pennylane import numpy as np
+from pennylane.circuit_graph import LayerData
 
 from .batch_transform import batch_transform
 
+_GEN_TO_CGEN = {
+    qml.PauliX: qml.CNOT,
+    qml.PauliY: qml.CY,
+    qml.PauliZ: qml.CZ,
+}
 
-def expand_fn(tape, approx="block-diag", diag_approx=None, allow_nonunitary=True):
+_OP_TO_CGEN = {
+    # PhaseShift is the same as RZ up to a global phase
+    qml.PhaseShift: qml.CZ,
+}

```

Comment:
This is because it maps the _operator_ to the controlled-generator operation whereas the one above maps _generators_ to controlled-generator operations.
The advantage of doing it this way - instead of putting all operators into `_OP_TO_CGEN` - is that a custom gate that has e.g. `custom_RX.generator = [qml.PauliX, 0.3]` will also be recognized :)
However, I did not want to go for any matrix arithmetic at this point, which would be necessary to detect `PhaseShift` being essentially generated by `qml.PauliZ`.

Reply:
got it! Hopefully this becomes simplified in the future with the proposed operator changes

Reply:
@dwierichs I recommend adding a small comment detailing your answer to the code 🙂 While reading this again, I had the exact same question, and was only reminded of the answer when un-collapsing this comment.